tm2/pkg/sdk/params/handler.go

```diff
@@ -0,0 +1,82 @@
+package params
+
+import (
+	"fmt"
+	"strings"
+
+	abci "github.com/gnolang/gno/tm2/pkg/bft/abci/types"
+	"github.com/gnolang/gno/tm2/pkg/sdk"
+	"github.com/gnolang/gno/tm2/pkg/std"
+)
+
+type paramsHandler struct {
+	params ParamsKeeper
+}
+
+func NewHandler(params ParamsKeeper) paramsHandler {
+	return paramsHandler{
+		params: params,
+	}
+}
+
+func (bh paramsHandler) Process(ctx sdk.Context, msg std.Msg) sdk.Result {
+	errMsg := fmt.Sprintf("unrecognized params message type: %T", msg)
+	return abciResult(std.ErrUnknownRequest(errMsg))
+}
+
+//----------------------------------------
+// Query
+
+func (bh paramsHandler) Query(ctx sdk.Context, req abci.RequestQuery) (res abci.ResponseQuery) {
+	switch secondPart(req.Path) {
+	case bh.params.prefix:
+		return bh.queryParam(ctx, req)
+	default:
+		res = sdk.ABCIResponseQueryFromError(
+			std.ErrUnknownRequest("unknown params query endpoint"))
+		return
+	}
+}
+
+// queryParam returns param for a key.
+func (bh paramsHandler) queryParam(ctx sdk.Context, req abci.RequestQuery) (res abci.ResponseQuery) {
+	// parse key from path.
+	key := thirdPartWithSlashes(req.Path)
+	if key == "" {
+		res = sdk.ABCIResponseQueryFromError(
+			std.ErrUnknownRequest("param key is empty"))
+	}
+
+	// XXX: validate?
+
+	val := bh.params.GetRaw(ctx, key)
```

Comment:
So to use this one you need to put a suffix like .bool, right?

Reply:
Yes, the suffix is part of the key. It is always required, even for type-specific helpers.